axiom/tests/test_txn.c

```diff
@@ -8098,6 +8098,126 @@ static void test_segment_record_error(void) {
   nr_txn_destroy(&txn);
 }
 
+static void test_segment_record_error_with_additional_params(void) {
+  nrapp_t app = {
+      .state = NR_APP_OK,
+      .limits = {
+        .analytics_events = NR_MAX_ANALYTIC_EVENTS,
+        .span_events = NR_DEFAULT_SPAN_EVENTS_MAX_SAMPLES_STORED,
+      },
+  };
+  nrtxnopt_t opts;
+  nr_segment_t* segment;
+  nrtxn_t* txn;
+
+  /* Setup transaction and segment */
+  nr_memset(&opts, 0, sizeof(opts));
+  opts.distributed_tracing_enabled = 1;
+  opts.span_events_enabled = 1;
+  txn = nr_txn_begin(&app, &opts, NULL);
+  segment = nr_segment_start(txn, NULL, NULL);
+  nr_distributed_trace_set_sampled(txn->distributed_trace, true);
+  txn->options.allow_raw_exception_messages = 1;
+
+  /* No error attributes added if error collection isn't enabled */
+  txn->options.err_enabled = 0;
+  nr_txn_record_error_with_additional_attributes(
+      txn, 1, true, "low priority message", "low priority class", "random.php",
+      150, "random context", 256, "[\"A\",\"B\"]");
+  tlib_pass_if_null("No segment error created", segment->error);
```

Comment:
Shouldn't `txn->error` also be checked that no error was created?

Reply:
Good point, added: [42a7a81](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/42a7a81482af738ad7a4602ad9babe4cab6becaa)